interview-questions.md

```diff
@@ -7,81 +7,83 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 1. Why would you use the method `super()`?
 
-  Your answer:
+  Your answer: Super is placed in the constructor of a child's React component so that it can have access to its parent's props.
 
-  Researched answer:
+  Researched answer: From the MDN web docs: the super key word is used to access and call functions on an object's parent. Additionally, the super keyword must be used before THIS can be used in the constructor. (I think I sort of understand this, but I need to do more React tutorials.)
 
 
 
 2. What is the DOM? What is the virtual DOM?
 
-  Your answer:
+  Your answer: DOM is the Document Object Model. The virtual DOM is the blueprint of the current DOM that is being rendered. React can update the currently loaded webpage without reloading because it only rerenders the portions of the page where the virtual DOM and the current DOM differ.
 
-  Researched answer:
+  Researched answer:DOM is a cross-platform and language independent interface that treats an XML or HTML document as a tree structure which allows a developer to change the structure, style or content of a document. The virtual DOM is an ideal representation of a UI that is synced with the "real" DOM by a library.
 
 
 
 3. What is a higher-order function?
 
-  Your answer:
+  Your answer: A higher order function is a function that either takes a function as an argument or returns a function when finished.
 
-  Researched answer:
+  Researched answer: Hey, I remembered the first sentence of that syllabus section almost verbatim. Go me.
 
 
 
 4. What is JSX?
 
-  Your answer:
+  Your answer: JSX is JavaScript Xomething. (Probably not Xtreme. Maybe XML?) It is a combination of JavaScript and XML (which looks a lot like HTML). It allows React to inject and execute JavaScript code in the layout of a website.
```

Comment:
You got it right with the XML part :)  Additionally, JSX is not strict to React! It's used in other places as well!